packages/mattermost-redux/src/reducers/entities/worktemplates.ts

```diff
@@ -5,39 +5,36 @@ import {combineReducers} from 'redux';
 
 import {GenericAction} from 'mattermost-redux/types/actions';
 import {WorkTemplatesType} from 'mattermost-redux/action_types';
+
 import {Category, WorkTemplate} from '@mattermost/types/worktemplates';
 
-function categories(state: Category[] = [], action: GenericAction) {
+function categories(state: Category[] = [], action: GenericAction): Category[] {
     switch (action.type) {
-    case WorkTemplatesType.RECEIVED_DEV_WORKTEMPLATES: {
-        const data = action.data as WorkTemplate[];
-        return data.
-            map((template) => template.category).
-            filter((category, index, self) => self.indexOf(category) === index);
+    case WorkTemplatesType.RECEIVED_WORK_TEMPLATE_CATEGORIES: {
+        const nextState = [...state];
+        for (const category of action.data) {
```

Comment:
question, is this loop necessary? not sure if just returning `action.data` would work